tools/cryptogenotyper/cryptogenotyper.xml

```diff
@@ -14,66 +14,56 @@
+        #if $db
           --databasefile $ref_file
         #end if
      #else
-        ln -s "${primers.abi_input}" "${primers.abi_input.element_identifier}" &&
-        cryptogenotyper -i "./${primers.abi_input.element_identifier}" -m "$marker" -t "$primers.seqtype" -f "" -r ""
-        #if $reference.ref == "no"
+        ln -s '${primers.abi_input}' '${primers.abi_input.element_identifier}' &&
+        cryptogenotyper -i './${primers.abi_input.element_identifier}' -m '$marker' -t '$primers.seqtype'
+        #if $db
           --databasefile $ref_file
         #end if
      #end if
-     #if $outputheader == "no"
+     #if $outputheader
            --noheaderline
      #end if
-     -o "result";
+     -o 'result'
      ]]>
   </command>
   <inputs>
       <param name="marker" type="select" label="Marker">
           <option value="18S">SSU rRNA</option>
           <option value="gp60">gp60</option>
       </param>
-      <conditional name="reference">
-        <param name="ref" type="select" label="Use default reference file?">
-            <option value="yes">Yes</option>
-            <option value="no">No</option>
-        </param>
-        <when value="yes">
-        </when>
-        <when value="no">
-            <param name="db" type="data" format="fasta" label="Reference Database File:" help=".fa fasta file type"/>
-        </when>
-      </conditional>
+      <param name="db" type="data" optional="true" format="fasta" label="Reference Database File (optional):"/>
       <conditional name="primers">
         <param name="seqtype" type="select" label="Type of Sequences">
             <option value="forward">Forward Only</option>
             <option value="reverse">Reverse Only</option>
             <option selected="true" value="contig">Contig</option>
         </param>
         <when value="contig">
-          <param name="abi_input" type="data_collection" collection_type="paired" format="ab1" label="Paired Sequencing File(s)" help=".ab1 file type"/>
+          <param name="abi_input" type="data_collection" collection_type="paired" format="ab1" label="Paired Sequencing File(s)"/>
         </when>
         <when value="forward">
-          <param name="abi_input" type="data" format="ab1" label="Forward Sequencing File(s)" help=".ab1 file type"/>
+          <param name="abi_input" type="data" format="ab1" label="Forward Sequencing File(s)"/>
         </when>
         <when value="reverse">
-          <param name="abi_input" type="data" format="ab1" label="Reverse Sequencing File(s)" help=".ab1 file type"/>
+          <param name="abi_input" type="data" format="ab1" label="Reverse Sequencing File(s)"/>
         </when>
      </conditional>
-     <param name="outputheader" type="select" value="yes" label="Output header line in the report?">
-        <option value="yes">Yes</option>
-        <option value="no">No</option>
+     <param name="outputheader" type="boolean" checked="true" label="Output header line in the report?">
```

Comment:
interesting, didn't know this is possible. You can also use 

`<param name="outputheader" type="boolean" truevalue="--noheaderline" falsevalue="" checked="true" label="Output header line in the report?" />`